kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -0,0 +1,746 @@
+            KRATOS_ERROR_IF(r_aux_geom.GetGeometryFamily() != GeometryData::Kratos_Triangle) <<
+                "In 2D the element type is expected to be a triangle." << std::endl;
+        } else if(dim == 3) {
+            KRATOS_ERROR_IF(r_aux_geom.GetGeometryFamily() != GeometryData::Kratos_Tetrahedra) <<
+                "In 3D the element type is expected to be a tetrahedron" << std::endl;
+        } else {
+            KRATOS_ERROR << "Wrong geometry Dimension(). Expected 2 or 3 and obtained: " << dim;
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    /// Destructor.
+    ~CalculateEmbeddedNodalVariableFromSkinProcess() override
+    {
+        Model& current_model = mrBaseModelPart.GetModel();
+        if(current_model.HasModelPart(mAuxModelPartName)) {
+            current_model.DeleteModelPart(mAuxModelPartName);
+        }
+    };
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override
+    {
+        KRATOS_TRY;
+
+        // Generate an auxilary model part and populate it by elements of type DistanceCalculationElementSimplex
+        this->GenerateIntersectedEdgesElementsModelPart();
+
+        // Set the linear strategy to solve the regression problem
+        this->SetLinearStrategy();
+
+        // Solve the regression problem
+        mpSolvingStrategy->Solve();
+
+        // Move the obtained values to the user-defined variable
+        this->SetObtainedEmbeddedNodalValues();
+
+        // Free the memory
+        this->Clear();
+
+        KRATOS_CATCH("")
+    }
+
+    virtual void Clear()
+    {
+        Model& current_model = mrBaseModelPart.GetModel();
+        ModelPart& r_distance_model_part = current_model.GetModelPart( mAuxModelPartName );
+        r_distance_model_part.Nodes().clear();
+        r_distance_model_part.Elements().clear();
+        r_distance_model_part.Conditions().clear();
+
+        mpSolvingStrategy->Clear();
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "CalculateEmbeddedNodalVariableFromSkinProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "CalculateEmbeddedNodalVariableFromSkinProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    const std::string mLevelSetType;
+    const std::string mAuxModelPartName;
+
+    ModelPart& mrBaseModelPart;
+    ModelPart& mrSkinModelPart;
```

Comment:
can they be `const`?

Reply:
Nope. I tried but there are some methods in the model part class that are not `const` (at some point we should take care of this).

Reply:
could you please make an issue for this? Maybe even with the list of functions that require a `const` version
I think it would be helpful to eventually solve this?

Btw in the vtk-output we use a `const ModelPart`, so I think there are not too many functions left that don't have a `const` counterpart